api/server/app.cc

```diff
@@ -494,38 +520,51 @@ void app::SetCgroup() {
     std::string setStr;
     std::string unsetOtherStr;
     std::ostringstream oss;
-
-    oss << app::config.tid;
-    setStr = "echo " + oss.str() + " > " + SrcCgroup() + "/butterfly/tasks";
-    unsetOtherStr = "grep -v " + oss.str() + " " + SrcCgroup() +
-                    "/butterfly/tasks | while read ligne; do echo $ligne > " +
-                    SrcCgroup() + "/tasks ; done";
-
-    BASH(setStr) {
-        LOG_WARNING_("can't set cgroup pid");
+    int check = 0;
+    std::string fileToWrite = std::string(SrcCgroup()) + "/butterfly/tasks";
+    check = setCgroups(oss.str(), fileToWrite);
+    if (!check){
+	LOG_WARNING_("can't set cgroup pid");
     }
-    BASH(unsetOtherStr) {
-        LOG_WARNING_("can't properly set cgroup pid");
+    std::ifstream ifs(oss.str());
+    std::ofstream ofs(fileToWrite);
+    std::string word, ligne;
+    while (ifs>>word){
+	if (oss.str().compare(word)){
+	    LOG_WARNING_("can't properly set cgroup pid");
+	    return ;
+	} else {
+	    ofs<<word ;
+	}
     }
 }
 
 void app::DestroyCgroup() {
     if (!SrcCgroup())
         return;
-    BASH("cat " + std::string(SrcCgroup()) +
-         "/butterfly/tasks | while read ligne; do echo $ligne > " +
-         SrcCgroup() + "/bin/bash /tasks ; done") {
-        LOG_WARNING_("can't unset task from butterfly cgroup");
+
+    std::string fileToRead = std::string(SrcCgroup()) + "/butterfly/tasks";
+    std::string fileToWrite = std::string(SrcCgroup()) + "/bin/bash /tasks";
+    std::ifstream in(fileToRead);
+    std::ofstream out(fileToWrite);
+    if ((!in.is_open()) || (!out.is_open())){
+	LOG_WARNING_("can't set cgroup priority");
+    } else {
+        std::string s;
+        if (getline(in, s)){
+            out << s << std::endl;
+	}
     }
-    BASH("rmdir " + std::string(SrcCgroup()) + "/butterfly") {
-        LOG_WARNING_("can't destroy cgroup");
+
+    int check = 0;
+    std::string dirToRemove = std::string(SrcCgroup()) + "/butterfly";
+    check = rmdir(dirToRemove.c_str());
+    if (!check){
+	LOG_WARNING_("can't destroy cgroup");
     }
 }
 
-#undef BASH
-
-int
-main(int argc, char *argv[]) {
+int main(int argc, char *argv[]) {
```

Comment:
well, that might be right, bit that's unrelated to this commit, so if you want to push that fix, please do it in a separated commit.